spark/v3.1/build.gradle

```diff
@@ -152,8 +152,8 @@ project(":iceberg-spark:iceberg-spark-extensions-3.1_2.12") {
 
     // Required because we remove antlr plugin dependencies from the compile configuration, see note above
     // We shade this in Spark3 Runtime to avoid issues with Spark's Antlr Runtime
-    runtimeOnly "org.antlr:antlr4-runtime:4.7.1"
-    antlr "org.antlr:antlr4:4.7.1"
+    runtimeOnly "org.antlr:antlr4-runtime:4.8"
+    antlr "org.antlr:antlr4:4.8"
```

Comment:
It seems like according to the comments in the build file that this should be shaded.

I'm guessing that the two still don't play nicely together? We might want to update some of the notes about shading the antlr runtime, as the comments make it seem like we don't need to worry about version conflicts with Spark (but the issue indicates we likely do).

Reply:
Yeah I wouldn't think this would be an issue except in the unshaded build, that said this is fine to do

Reply:
I don't think that we shade antlr. I saw a conflict warning about this as well. @kbendick, what comment made you think that we are shading antlr?

Reply:
I guess it’s just in the runtime jar that we shade. The comment above and then this one: https://github.com/apache/iceberg/blob/25150e5ceb7c84a6dae773a12a5061edb55f2c8b/spark/v3.1/build.gradle#L253

Reply:
I remember we shaded antlr because we choose to make spark3-runtime support both spark 3.0/3.1, and they bundle the different version of antlr, since we split spark runtime jar by minor version, we can exclude antlr and avoid shading.

Reply:
Thanks for the context, @pan3793! I forgot that we had started shading Antlr